internal/unionstore/art/art.go

```diff
@@ -0,0 +1,644 @@
+package art
+
+import (
+	"fmt"
+	"math"
+	"sync"
+
+	tikverr "github.com/tikv/client-go/v2/error"
+	"github.com/tikv/client-go/v2/kv"
+)
+
+var tombstone = []byte{}
+
+type Art struct {
+	sync.RWMutex
+	skipMutex       bool
+	allocator       artAllocator
+	root            artNode
+	stages          []ARTCheckpoint
+	vlogInvalid     bool
+	dirty           bool
+	entrySizeLimit  uint64
+	bufferSizeLimit uint64
+	len             int
+	size            int
+}
+
+func New() *Art {
+	var t Art
+	t.root = nullArtNode
+	t.stages = make([]ARTCheckpoint, 0, 2)
+	t.entrySizeLimit = math.MaxUint64
+	t.bufferSizeLimit = math.MaxUint64
+	//t.allocator.init()
+	t.allocator.nodeAllocator.freeNode4 = make([]nodeAddr, 0, 1<<4)
+	t.allocator.nodeAllocator.freeNode16 = make([]nodeAddr, 0, 1<<3)
+	t.allocator.nodeAllocator.freeNode48 = make([]nodeAddr, 0, 1<<2)
+	t.allocator.nodeAllocator.freeNode256 = make([]nodeAddr, 0, 1)
```

Comment:
Why do they have different capacity?
Can we assure the length of these slices are always limited, i.e. there won't be an unlimited-long slice of free nodes?

Reply:
`freeNode256` is not required actually, I'll remove it.

Don't worry about the unlimited-long slice of free nodes, it's reused [here](https://github.com/you06/client-go/blob/90e7e39c64bf24309d6a8aeb401e5519ffa1dbe2/internal/unionstore/art/arena.go#L107-L109).

When a node4 is grown to node16, the old one will be put into the `freeNode4` and the next `allocNode4` can reuse it, I support the node with less capacity is allocated with higher frequency and have more chance to be recycled, so the free space of node4 is the longest by default.